src/Microsoft.Extensions.Primitives/StringSegment.cs

```diff
@@ -297,6 +345,17 @@ public string Substring(int offset, int length)
 
         /// <summary>
         /// Retrieves a <see cref="StringSegment"/> that represents a substring from this <see cref="StringSegment"/>.
+        /// The <see cref="StringSegment"/> starts at the position specified by <paramref name="offset"/>.
+        /// </summary>
+        /// <param name="offset">The zero-based starting character position of a substring in this <see cref="StringSegment"/>.</param>
+        /// <returns>A <see cref="StringSegment"/> that is equivalent to the that begins at <paramref name="offset"/> in this <see cref="StringSegment"/></returns>
```

Comment:
A substring of this instance of `StringSegment` that begins at `offset`? Doesn't sound right as it is now